Design/Instrument-2.0/Detector_Scan_NeXus_Saving.md

```diff
@@ -0,0 +1,129 @@
+
+## Current Implementation
+
+Spectra to detector mapping is stored in `mantid_workspace_1/instrument/detector`. There are five fields, `spectra` which provides the spectrum numbers, `detector_count` which provides the number of detectors for each spectra, `detector_list` which provides a list of all the detectors, `detector_index`, which gives the entry point into `detector_list` for a given spectra and `detector_positions` which, for unknown reasons, gives the R, &theta; and &phi; coordinates of every detector.
+
+The 'true' source for the detector positions and rotations is in `mantid_workspace_1/instrument/instrument_parameter_map/data`. They are only stored here if they differ from the base instrument.
+
+
+## Implementation Ideas
+
+### New NeXus Instrument Format
+
+_Based on discussion with Matt Jones_
+
+Work has been done for the ESS for a new instrument format in the NeXus files. The approach in the new format, effectively a replacement for the old XML IDF, contains detectors (components in Mantid terminology) with pixels, for example a detector tube or bank with sub-pixels.
+
+See the example for the NeXus hierarchy below. For a component with pixels in 3-dimensions the detector numbers are given, and then the offsets are given in x, y and z. Not all of the offsets are required, for example a tube might only require the `y_pixel_offset` entry.
+
+```
+raw_data_1
+├───instrument
+│   ├───detector_1
+│   │   ├───detector_number (size is number of detectors)
+│   │   ├───x_pixel_offset (size is number of detectors)
+│   │   ├───y_pixel_offset (size is number of detectors)
+│   │   ├───z_pixel_offset (size is number of detectors)
+│   │   ├───...
+│   ├───detector_2
+│   ├───...
+├───sample
+```
+
+The suggested method for dealing with detector scans is to add entries as `NXTransformation`s to the detector entry. This is shown in the example hierarchy below. For each position and rotation axis a separate entry is required within the detector, such as `position_scan_x`, `position_scan_y`, `position_scan_z`, `rotation_scan_x`, `rotation_scan_y` and `rotation_scan_z`.
+
+```
+raw_data_1
+├───instrument
+│   ├───detector_1
+│   │   ├───detector_number
+│   │   ├───x_pixel_offset
+│   │   ├───y_pixel_offset
+│   │   ├───z_pixel_offset
+│   │   ├───position_scan_x (attribute 'vector' for transformation, e.g. (1., 0., 0.))
+│   │   │   ├───scan_log
+│   │   │       ├───time (contains the time for each entry in value)
+│   │   │       ├───value (single number per time entry, describing the distance along 'vector')
+│   │   ├───position_scan_...
+│   ├───detector_2
+│   ├───...
+├───sample
+```
+
+An open question here, especially when considering conversion from the old IDFs, is how to define the component used for `detector_X`. Technically this could be anything from the whole instrument to individual pixels, but logically it should be something like a tube. The only requirement is that a detector has pixels with a fixed offset at all times.
```

Comment:
Can detectors commonly be described in such a regular way? How would this deal with calibration or mechanical tolerances? Would that require having scan information for all pixels individually?

What I am saying is that maybe we want a way to have per-bank scanning in Nexus but allow pixels within the bank to be non-equidistant? Does that make sense?

Reply:
The above isn't 100% complete. I think what you want @SimonHeybrock is supported, but we should improve the example. @matthew-d-jones is in touch.

Reply:
In the NeXus instrument there would be no requirement for pixels to be equidistant, offsets for each pixel from the centre are given in 3 dimensions. I'm not sure for rotations though.